### PR TITLE
#182 customizable nonfatal

### PR DESCRIPTION
**Pull request by pshirshov**
A simple fix for #182

### CODE REVIEW

File: core/jvm/src/main/scala/scalaz/zio/RTS.scala [hunk 1]
```diff
@@ -27,7 +27,7 @@ trait RTS {
 
   final def unsafeRunAsync[E, A](io: IO[E, A])(k: Callback[E, A]): Unit = {
     val context = new FiberContext[E, A](this, defaultHandler)
-    context.evaluate(io)
+    context.evaluate(this, io)
```
Comment (regiskuckaertz) [inline core/jvm/src/main/scala/scalaz/zio/RTS.scala]: The runtime system is already passed at construction time (see the line just above), so you don't need to thread it everywhere.

**Comment by pshirshov** [comment]
@regiskuckaertz : P/R updated.

**Comment by jmcardon** [comment]
I'm :-1: on this pr. See #182 for discussion.

**Comment by pshirshov** [comment]
Answered you there. This doesn't make things worse. Could you propose any sane scenario where this would be a problem? But at the same time it gives you bit more flexibility - in a rare case when you need to catch everything - you may do it. Though this option is hidden and can't be used accidentally.

**Comment by jdegoes** [comment]
I'm going to merge because the only change this makes is moving `nonFatal` into RTS, which is arguably a better place. It also adds the ability to change `nonFatal` to make it consistent with Scala's notion of `NonFatal`, which is different than Monix' notion of `NonFatal`, which is different than Cats Effect notion of `NonFatal` (unless that's been fixed).

**Comment by pshirshov** [comment]
Thanks!

**Pull request merged**